Source/JavaScriptCore/bytecode/ArrayAllocationProfile.cpp

```diff
@@ -53,19 +53,20 @@ void ArrayAllocationProfile::updateProfile()
     // So for now, we update the allocation profile only from the main thread.
     
     ASSERT(!isCompilationThread());
-    JSArray* lastArray = std::exchange(m_lastArray, nullptr);
+    Storage storage = std::exchange(m_storage, Storage(nullptr, m_storage.type()));
+    JSArray* lastArray = storage.pointer();
     if (!lastArray)
         return;
     if (LIKELY(Options::useArrayAllocationProfiling())) {
         // The basic model here is that we will upgrade ourselves to whatever the CoW version of lastArray is except ArrayStorage since we don't have CoW ArrayStorage.
-        IndexingType indexingType = leastUpperBoundOfIndexingTypes(m_currentIndexingType & IndexingTypeMask, lastArray->indexingType());
-        if (isCopyOnWrite(m_currentIndexingType)) {
+        IndexingType indexingType = leastUpperBoundOfIndexingTypes(current().indexingType() & IndexingTypeMask, lastArray->indexingType());
+        if (isCopyOnWrite(current().indexingType())) {
             if (indexingType > ArrayWithContiguous)
                 indexingType = ArrayWithContiguous;
             indexingType |= CopyOnWrite;
         }
-        m_currentIndexingType = indexingType;
-        m_largestSeenVectorLength = std::min(std::max(m_largestSeenVectorLength, lastArray->getVectorLength()), BASE_CONTIGUOUS_VECTOR_LEN_MAX);
+        unsigned largestSeenVectorLength = std::min(std::max(current().vectorLength(), lastArray->getVectorLength()), BASE_CONTIGUOUS_VECTOR_LEN_MAX);
+        m_storage.setType(IndexingTypeAndVectorLength(indexingType, largestSeenVectorLength));
```

Comment:
Let's put `static_assert(BASE_CONTIGUOUS_VECTOR_LEN_MAX < UINT8_MAX)` to ensure that vector-length is within uint8_t.

Reply:
The assertion already exists in `IndexingTypeAndVectorLength`